hasher-matcher-actioner/hmalib/lambdas/fetcher.py

```diff
@@ -315,23 +317,56 @@ def post_apply(self, updated: t.Dict = {}):
         table = dynamodb.Table(self.data_store_table)
 
         for update in updated.values():
-            row = update.as_csv_row()
+            row: t.List[str] = update.as_csv_row()
             # example row format: ('<raw_indicator>', '<indicator-id>', '<descriptor-id>', '<time added>', '<space-separated-tags>')
             # e.g (10736405276340','096a6f9...064f', '1234567890', '2020-07-31T18:47:45+0000', 'true_positive hma_test')
-            if PDQSignalMetadata(
-                signal_id=int(row[1]),
+            new_tags = row[4].split(" ") if row[4] else []
+
+            # Figure out if we have a new opinion about this indicator and clear out a pending change if so
+            # If this is a new indicator without metadata we also should have no pending opinion about it
+            new_pending_opinion_change = PendingOpinionChange.NONE
+            for signal in PDQSignalMetadata.get_from_signal(
```

Comment:
Unfortunately the way `get_from_signal` works (`query`) is a fair amount slower than way we'd implement `get_from_signal_and_ds_id` (`get_item`). 
This is bad because for every updated item we run the query. If you want help with writing something `get_from_signal_and_ds_id` tomorrow morning or you'd just like me to write it (more than happy to because it's a gap I left in that model when I created it) let me know but I do think it is blocking